flink-connectors/flink-connector-kafka-0.9/src/test/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThreadTest.java

```diff
@@ -875,109 +878,194 @@ void reassignPartitions(List<KafkaTopicPartitionState<TopicPartition>> newPartit
+		}
 
+		@Override
+		public void seek(TopicPartition partition, long position) {
+			if (!mockConsumerAssignmentAndPosition.containsKey(partition)) {
+				throw new RuntimeException("the current mock assignment does not contain partition " + partition);
+			} else {
+				mockConsumerAssignmentAndPosition.put(partition, position);
+			}
+		}
+
+		@Override
+		public void seekToBeginning(TopicPartition... partitions) {
+			for (TopicPartition partition : partitions) {
 				if (!mockConsumerAssignmentAndPosition.containsKey(partition)) {
-					throw new Exception("the current mock assignment does not contain partition " + partition);
+					throw new RuntimeException("the current mock assignment does not contain partition " + partition);
 				} else {
 					Long mockRetrievedPosition = mockRetrievedPositions.get(partition);
 					if (mockRetrievedPosition == null) {
-						throw new Exception("mock consumer needed to retrieve a position, but no value was provided in the mock values for retrieval");
+						throw new RuntimeException("mock consumer needed to retrieve a position, but no value was provided in the mock values for retrieval");
 					} else {
 						mockConsumerAssignmentAndPosition.put(partition, mockRetrievedPositions.get(partition));
 					}
 				}
-				return null;
 			}
-		}).when(mockConsumer).seekToBeginning(any(TopicPartition.class));
-
-		doAnswer(new Answer() {
-			@Override
-			public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
-				TopicPartition partition = invocationOnMock.getArgument(0);
+		}
 
+		@Override
+		public void seekToEnd(TopicPartition... partitions) {
+			for (TopicPartition partition : partitions) {
 				if (!mockConsumerAssignmentAndPosition.containsKey(partition)) {
-					throw new Exception("the current mock assignment does not contain partition " + partition);
+					throw new RuntimeException("the current mock assignment does not contain partition " + partition);
 				} else {
 					Long mockRetrievedPosition = mockRetrievedPositions.get(partition);
 					if (mockRetrievedPosition == null) {
-						throw new Exception("mock consumer needed to retrieve a position, but no value was provided in the mock values for retrieval");
+						throw new RuntimeException("mock consumer needed to retrieve a position, but no value was provided in the mock values for retrieval");
 					} else {
 						mockConsumerAssignmentAndPosition.put(partition, mockRetrievedPositions.get(partition));
 					}
 				}
-				return null;
 			}
-		}).when(mockConsumer).seekToEnd(any(TopicPartition.class));
+		}
+
+		@Override
+		public long position(TopicPartition topicPartition) {
+			if (!earlyWakeup) {
+				return mockConsumerAssignmentAndPosition.get(topicPartition);
+			} else {
+				throw new WakeupException();
+			}
+		}
 
-		return mockConsumer;
+		@Override
+		public OffsetAndMetadata committed(TopicPartition topicPartition) {
+			return null;
```

Comment:
Should we throw `UnsupportedOperationException` here in order to avoid NPE somewhere else in the code?

Reply:
Tests still passes with all methods that return something throwing an exception instead, so we'll go with that.